lib/migrate/Migrator.js

```diff
@@ -80,12 +80,16 @@ class Migrator {
 
     const transactionForAll =
       !this.config.disableTransactions &&
-      !migrations.some((migration) => {
-        const migrationContents = this.config.migrationSource.getMigration(
-          migration
-        );
-        return !this._useTransaction(migrationContents);
-      });
+      !(
+        await Promise.all(
+          migrations.map(async (migration) => {
+            const migrationContents = await this.config.migrationSource.getMigration(
+              migration
+            );
+            return !this._useTransaction(migrationContents);
+          })
+        )
+      ).some((isTransactionUsed) => isTransactionUsed);
```

Comment:
were the changes in `Migrator` intended in this PR?

Reply:
Okay I can see more details in https://github.com/knex/knex/pull/4296

Reply:
@falkenhawk Did that cause some breaking changes for you?

Reply:
I can't tell yet, I noticed this while I was looking through the changes after I got notified by renovatebot about the new knex version 0.21.18... and not having this explicitly listed in changelog confused me and stopped me from upgrading, as it seemed to me to be something extra than just a fix for #4296 - and I could not afford breaking things at that moment 😅

If I decide to upgrade in the following days and will run into some issues, I will let you know then.

Reply:
It should behave same way for the end user in most cases, and also handle ESM case better, see https://github.com/knex/knex/pull/4296#issuecomment-779330504 for the reason of more methods becoming async. If you are not using ESM, this upgrade won't do anything useful for you.